shoop/admin/modules/products/views/forms.py

```diff
@@ -178,10 +178,19 @@ def __init__(self, **kwargs):
         self.fields["file"].widget = MediaChoiceWidget()  # Filer misimplemented the field; we need to do this manually.
         self.fields["file"].required = True
 
-        if len(self.allowed_media_kinds) == 1:
-            # only one media kind given, no point showing the dropdown
+        if self.allowed_media_kinds:
+            if len(self.allowed_media_kinds) == 1:
+                # only one media kind given, no point showing the dropdown
+                self.fields["kind"].widget = forms.HiddenInput()
+            else:
```

Comment:
You should filter the choices even if the input is hidden; `choices` affects the validation as well.